src/js/edu-benefits/definitions/veteranId.js

```diff
@@ -0,0 +1,44 @@
+import _ from 'lodash/fp';
+
+import * as ssn from '../../common/schemaform/definitions/ssn';
+
+// TODO: move vaFileNumber out of 1995 and into a shared backend definition
+import fullSchema1995 from 'vets-json-schema/dist/change-of-program-schema.json';
+const {
+  vaFileNumber
+} = fullSchema1995.definitions;
+
+
+export const uiSchema = {
+  veteranSocialSecurityNumber: _.assign(ssn.uiSchema, {
+    'ui:required': (form) => !form.veteranId['view:noSSN']
```

Comment:
When changing from `veteranId` to `view:veteranId`, this (and line 23 below) needs to change too.
```js
!form['view:veteranId']['view:noSSN']
```